src/packet/public_key_encrypted_session_key.js

```diff
@@ -110,25 +110,45 @@ class PublicKeyEncryptedSessionKeyPacket {
   }
 
   /**
-   * Decrypts the session key (only for public key encrypted session key
-   * packets (tag 1)
+   * Decrypts the session key (only for public key encrypted session key packets (tag 1)
    * @param {SecretKeyPacket} key - decrypted private key
-   * @throws {Error} if decryption failed
+   * @param {Object} [randomSessionKey] - Bogus session key to use in case of sensitive decryption error, or if the decrypted session key is of a different type/size.
+   *                                      This is needed for constant-time processing. Expected object of the form: { sessionKey: Uint8Array, sessionKeyAlgorithm: enums.symmetric }
+   * @throws {Error} if decryption failed, unless `randomSessionKey` is given
    * @async
    */
-  async decrypt(key) {
+  async decrypt(key, randomSessionKey) {
     // check that session key algo matches the secret key algo
     if (this.publicKeyAlgorithm !== key.algorithm) {
       throw new Error('Decryption error');
     }
-    const decoded = await crypto.publicKeyDecrypt(this.publicKeyAlgorithm, key.publicParams, key.privateParams, this.encrypted, key.getFingerprintBytes());
+
+    const randomPayload = randomSessionKey ? util.concatUint8Array([
+      new Uint8Array([randomSessionKey.sessionKeyAlgorithm]),
+      randomSessionKey.sessionKey,
+      util.writeChecksum(randomSessionKey.sessionKey)
+    ]) : null;
+    const decoded = await crypto.publicKeyDecrypt(this.publicKeyAlgorithm, key.publicParams, key.privateParams, this.encrypted, key.getFingerprintBytes(), randomPayload);
     const checksum = decoded.subarray(decoded.length - 2);
     const sessionKey = decoded.subarray(1, decoded.length - 2);
-    if (!util.equalsUint8Array(checksum, util.writeChecksum(sessionKey))) {
-      throw new Error('Decryption error');
+    const symmetricAlgoByte = decoded[0];
+    const isValidChecksum = util.equalsUint8Array(checksum, util.writeChecksum(sessionKey));
```

Comment:
`equalsUint8Array` is not constant-time - since both are guaranteed to be two bytes, I would just hardcode an equality check for those two bytes here